client/lib/titan/get-max-titan-mailbox-count.js

```diff
@@ -0,0 +1,3 @@
+export function getMaxTitanMailboxCount( domain ) {
+	return domain.titanMailSubscription?.maximumMailboxCount ?? 1;
```

Comment:
If this is operating on the Redux store (as I'm assuming based on the properties), I think putting this in selectors is a better fit. Possibly here: https://github.com/Automattic/wp-calypso/blob/trunk/client/state/sites/domains/selectors.js - though, the way domains are mixed in the sites thing has always been confusing for me and maybe there's a better place in that tree, especially for Titan selectors.

I think in domains code we've been misusing the `selectedDomain` pattern - instead, the cleaner way seems to be using selectors to inject only the properties we need based on the current state. So instead of `if ( hasTitanMailWithUs( selectedDomain ) ) {`, it should be `if ( this.props.hasTitanMailWithUs() ) {`, where `this.props.hasTitanMailWithUs`'s value comes from calling `hasTitanMailWithUs( selectedDomain`, but inside `connect`.

Long-term, this should lead to less re-rendering, as the component is only getting the changes it needs, not when the whole `selectedDomain` object changes every time.

Sorry for the rant, this is definitely not your fault nor was this anti-pattern introduced lately - I did not even realize how we were abusing `selectedDomain` until recently.

I hope that we can at least try to keep the Titan codebase a bit better organized, hence my suggestion 🙇‍♂️

Since @sirbrillig is already reviewing this code, I'd love to hear his expert opinion as well - I could be wrong here (again) :D Or there might be an even better way of handling this.

Reply:
Agreed. I updated some of the code based on your feedback, so I'd appreciate another look here 🙇

Reply:
Thanks for the changes! The move to `connect` for stuff like `getMaxTitanMailboxCount` and `hasTitanMailWithUs` is what I had in mind. However, they should _not_ take `selectedDomain`, but instead the whole `state`. And be moved to somewhere in `/client/state/selectors`. So, for example, `getMaxTitanMailboxCount`'s declaration would be `getMaxTitanMailboxCount( state, domainName )` and it'd use other selectors to get what it needs:

```
const selectedSiteId = getSelectedSiteId( state );
const selectedDomain = getSelectedDomain( {
			getDomainsBySiteId( state, selectedSiteId ),
			domainName,
		} );
return selectedDomain.titanMailSubscription?.maximumMailboxCount ?? 1;
```

But this is just my understanding how this should be done - I could be very much wrong. But it feels like using the selectedDomain pattern seems to be leading to a bit tightly coupled code. If we ever want to decouple Titan and the domain (store it in separate places in the tree), we'd have to update all these components to not fetch the selected domain, etc. But if we hide that behind the selector, it'd just be a matter of updating the selector to take the data from a different subtree, etc.

Feel free to leave it as-is, merge (if this PR is blocking further changes) and we can see what's Payton's expert opinion. Just want to take this opportunity to curb some patterns we've been using blindly for a long time which might be hurting us long term 🙇‍♂️

Reply:
Agreed. For now I will merge the PR, but will definitely iterate on this soon.